src/components/CurrentUser/CurrentUser.jsx

```diff
@@ -1,12 +1,74 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import './CurrentUser.scss';
 
-export const CurrentUser = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+import { getUser } from '../../api/api';
+import { currentUserType } from '../../types';
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+export class CurrentUser extends PureComponent {
+  state = {
+    user: {},
+  }
+
+  componentDidMount() {
+    this.updateUser(this.props.userId);
+  }
+
+  componentDidUpdate() {
+    const { userId } = this.props;
+
+    this.updateUser(userId);
+  }
+
+  updateUser(newUserId) {
+    getUser(newUserId)
+      .then((newUser) => {
+        this.setState({
+          user: newUser,
+        });
+      })
+      .catch(() => {
+        this.setState({
+          user: {},
+        });
+      });
+  }
+
+  render() {
+    const { user } = this.state;
+    const { onButtonClick } = this.props;
+
+    return (
+      <div className="CurrentUser">
+        {user
+          ? (
+            <>
+              <h2 className="CurrentUser__title">
+                <span>
+                  {`Selected user: ${user.id}`}
+                </span>
+              </h2>
+
+              <h3 className="CurrentUser__name">{user.name}</h3>
+              <p className="CurrentUser__email">{user.email}</p>
+              <p className="CurrentUser__phone">{user.phone}</p>
+
+              <button
+                type="button"
+                onClick={() => {
+                  onButtonClick();
+                }}
```

Comment:
pass just callback if you don't pass arguments

```suggestion
                onClick={selectUser}
```